### PR TITLE
Textbox: confirm close if text was changed

### PR DESCRIPTION
**Pull request by kravets-levko**
## What type of PR is this? (check all applicable)

- [x] Feature

## Description

When adding or editing a textbox it's very easy to accidentally close the dialog and lose all changes (by clicking outside of dialog). This PR solves this issue by showing a confirmation dialog when trying to close a textbox dialog with changes.

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

image

### CODE REVIEW

File: client/app/components/dashboards/TextboxDialog.jsx [hunk 1]
```diff
@@ -40,11 +40,31 @@ function TextboxDialog({ dialog, isNew, ...props }) {
     });
   }, [dialog, isNew, text]);
 
+  const confirmDialogDismiss = useCallback(() => {
+    const originalText = props.text;
+    if (text !== originalText) {
+      Modal.confirm({
+        title: "Confirm action",
```
Comment (gabrieldutra) [inline client/app/components/dashboards/TextboxDialog.jsx:47]:
```suggestion
        title: "Confirm Discard Changes",
```

File: client/app/components/dashboards/TextboxDialog.jsx [hunk 2]
```diff
@@ -40,11 +40,31 @@ function TextboxDialog({ dialog, isNew, ...props }) {
...
+        content: "Closing this dialog will discard all changes you've made. Are you sure?",
+        okText: "Yes, discard changes",
+        okType: "danger",
+        cancelText: "No, get me back",
```
Comment (gabrieldutra) [inline client/app/components/dashboards/TextboxDialog.jsx:51]:
```suggestion
        cancelText: "Cancel",
```
(Is this the default?)

**Comment by kravets-levko** [comment]
Thank you @gabrieldutra! I'm not good in texting :sweat_smile: 

image

File: client/app/components/dashboards/TextboxDialog.jsx [hunk 3]
```diff
@@ -40,11 +40,30 @@ function TextboxDialog({ dialog, isNew, ...props }) {
     });
   }, [dialog, isNew, text]);
 
+  const confirmDialogDismiss = useCallback(() => {
+    const originalText = props.text;
+    if (text !== originalText) {
+      Modal.confirm({
+        title: "Quit editing?",
+        content: "Changes you made so far will not be saved. Are you sure?",
+        okText: "Yes, discard changes",
```
Comment (gabrieldutra) [inline client/app/components/dashboards/TextboxDialog.jsx:49]:
```suggestion
        okText: "Yes, quit",
```

Lastly, just to make it aligned with the heading.

**Pull request merged**